### PR TITLE
Parallelize CI

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR:
 - Breaks up our tests into separate jobs to parallelize the work
 - Removes `test/test_cases.jl`, since the calls in here have been moved to the separate files
 - Remove GHA CI from bors (we'll still run on GHA, but passing CI doesn't depend on its success)
 - Adds a `test/Project.toml`

This PR also adds
```julia
if !haskey(ENV, "BUILDKITE")
    import Pkg
    Pkg.develop(Pkg.PackageSpec(; path = ...))
end
```
To the test scripts to ensure users don't accidentally fallback on the latest release of ClimaAtmos.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
bors try

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -8,9 +8,11 @@ steps:
   - label: "init :computer:"
     key: "init_cpu_env"
     command:
-      - "julia --project -e 'using Pkg; Pkg.instantiate(;verbose=true)'"
```
Comment (jakebolewski) [inline .buildkite/pipeline.yml:11]: Need to keep this in addition to the test project

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Pull request merged**